### DISCUSSION TITLE
Disconnect session returns error

### DISCUSSION THREAD

**Discussion by afcharliekong** [WalletConnect/walletconnect-monorepo#2682] [Developer Support] [answered]
### Issue

Whenever I tried to disconnect session via the code snippet below, I kept getting this error that could not be caught.

```
try {
      await this.web3wallet.disconnectSession({
        topic,
        reason: getSdkError("USER_DISCONNECTED"),
      });
    } catch (error) {
      console.log(error)
    }
```

The error returned is usually something like so

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/103051707/66300c86-6dcd-48cc-86fc-bfe23d7a73ee)

or this,

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/103051707/b94d038b-06aa-4219-a89f-d83e41415274)

This happens if I were to refresh the wallet page after establishing connection and try to disconnect after. I'm not sure if initialising a new instance of web3wallet is causing this issue or not.

### SDK Used

- "@walletconnect/utils": "2.8.1",
- "@walletconnect/web3wallet": "^1.8.0"

Any assistance is greatly appreciated.

**Comment by afcharliekong** [comment]
Bumping for visibility

**Comment by mqklin** [comment]
+1: `{context: 'core'} {context: 'core/expirer'} 'No matching key. expirer: topic:5394b64a82e83fba31731fb270c5c251aa9c993cc46d875e4a8fa60470b18a9b'`

**Comment by DamperX** [comment]
+1 The same issue

**Comment by glitch-txs** [comment, marked as answer]
Aswered [here](https://github.com/orgs/WalletConnect/discussions/3291). Seems to be an issue on the Wallet side.
